### PR TITLE
VRVU-33: add zScore calculator

### PR DESCRIPTION
**Pull request by jdolkowski**
https://sd-vxnaid.atlassian.net/browse/VRVU-25

For now zScore is a random value. Waiting for the calculation formula as mentioned in the ticket.

### CODE REVIEW

File: app/src/main/java/com/jnj/vaccinetracker/visit/screens/VisitDosingFragment.kt [hunk 1]
```diff
@@ -52,50 +57,66 @@ class VisitDosingFragment : BaseFragment(),
...
             if (!s.isNullOrEmpty()) {
                 viewModel.matchBarcodeManufacturer(s, resourcesWrapper)
-            }
-             else{
+            } else {
                 viewModel.setSelectedManufacturer("")
                 binding.dropdownManufacturer.clearListSelection()
-                }
+            }
+        }
```
Comment (sherrif10) [inline app/src/main/java/com/jnj/vaccinetracker/visit/screens/VisitDosingFragment.kt]: This looks good already, What are the conditions for the app to show obese.
Screenshot 2024-06-10 074102

Comment (jdolkowski) [reply app/src/main/java/com/jnj/vaccinetracker/visit/screens/VisitDosingFragment.kt]: The exact formula is not know yet. It shows random values for now.

File: app/src/main/java/com/jnj/vaccinetracker/visit/zscore/InputFilterMinMax.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            dstart: Int,
+            dend: Int
+    ): CharSequence? {
+        try {
+            val input = (dest.toString() + source.toString()).toInt()
+            if (isInRange(min, max, input)) {
+                return null
+            }
+        } catch (nfe: NumberFormatException) {
+            // Handle the exception
```
Comment (sherrif10) [inline app/src/main/java/com/jnj/vaccinetracker/visit/zscore/InputFilterMinMax.kt]: Are you planning to log an exception message, or adding something like `Log.e("InputFilterMinMax", NumberFormatException: ${nfe.message}")`

Comment (jdolkowski) [reply app/src/main/java/com/jnj/vaccinetracker/visit/zscore/InputFilterMinMax.kt]: good point

**Pull request merged**